packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -349,6 +351,59 @@ export const ObjectTestCases = {
     });
 
     this.assertText('T1T2F3');
+  },
+
+  ['@test setting the isTruthy property of an object proxy overrides its truthiness regardless of its content']() {
+    this.renderValues(
+      ObjectProxy.create({ content: {} }),
+      EmberObject.create({ content: false, isTruthy: true }),
+      ObjectProxy.create({ content: null })
+    );
+
+    this.assertText('T1T2F3');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('T1T2F3');
+
+    this.runTask(() => {
+      set(this.context, 'cond1.isTruthy', false);
+      set(this.context, 'cond2.isTruthy', false);
```

Comment:
ditto – we should not be setting `isTruthy` on the proxy, just change/set the content